common/workunit/workunit.cpp

```diff
@@ -14220,8 +14220,16 @@ extern WORKUNIT_API void addTimeStamp(IWorkUnit * wu, unsigned wfid, const char
     addTimeStamp(wu, getScopeType(scope), scope, kind, wfid);
 }
 
-static double getCostCpuHour()
+static double getCostCpuHour(const char * subComponentName = nullptr)
 {
+    if (!isEmptyString(subComponentName))
+    {
+        VStringBuffer subComponentSection("%s/cost/@perCpu", subComponentName);
+        double costCpuHour = getComponentConfigSP()->getPropReal(subComponentSection.str(), -1.0);
+        if (costCpuHour >= 0.0)
+            return costCpuHour;
+        // don't have sub component cost, drop-through to use component/global cost values
```

Comment:
I think this should be an error, and the templates should ensure that the sub section is generated.
i.e. after this PR, Thor should always generate a "eclagent", "manager", and, "worker" sections.
The legacy support for the old "cost" can be handled in the templates, allowing this code to be simplified/standardized on the new schema.

NB: the chart version must match the image version.

Reply:
Thor will have the cost section in the sub-section.   It is not necessary the case for other components such as eclagent and eclcc.  I've created Jira to make this change: https://hpccsystems.atlassian.net/browse/HPCC-35054